crates/starknet-devnet/tests/test_blocks_generation.rs

```diff
@@ -412,6 +418,64 @@ mod blocks_on_demand_tests {
         assert_eq!(get_contract_balance(&devnet, contract_address).await, expected_balance);
     }
 
+    #[tokio::test]
+    async fn blocks_on_interval() {
+        let devnet = BackgroundDevnet::spawn_with_additional_args(&["--block-generation-on", "1"])
+            .await
+            .expect("Could not start Devnet");
+
+        // wait 1 second
+        tokio::time::sleep(time::Duration::from_secs(1)).await;
+
+        let last_block = devnet.get_latest_block_with_tx_hashes().await.unwrap();
+
+        // first is genesis block, second block is generated instantly, third is generated after 1
+        // second
+        assert_eq!(last_block.block_number, 2);
+    }
+
+    #[tokio::test]
+    async fn blocks_on_interval_dump_and_load() {
+        let mode = "exit";
+        let dump_file = UniqueAutoDeletableFile::new(
+            ("blocks_on_interval_dump_and_load_".to_owned() + mode).as_str(),
+        );
+
+        let devnet_dump = BackgroundDevnet::spawn_with_additional_args(&[
+            "--dump-path",
+            &dump_file.path,
+            "--dump-on",
+            mode,
+            "--block-generation-on",
+            "1",
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        // wait 1 second
+        tokio::time::sleep(time::Duration::from_secs(1)).await;
+
+        let last_block = devnet_dump.get_latest_block_with_tx_hashes().await.unwrap();
+
+        // first is genesis block, second block is generated instantly, third is generated after 1
+        // second
+        assert_eq!(last_block.block_number, 2);
+
+        send_ctrl_c_signal_and_wait(&devnet_dump.process).await;
+
+        let devnet_load = BackgroundDevnet::spawn_with_additional_args(&[
+            "--dump-path",
+            dump_file.path.as_str(),
+            "--dump-on",
+            mode,
+        ])
+        .await
+        .expect("Could not start Devnet");
+
+        let last_block_load = devnet_load.get_latest_block_with_tx_hashes().await.unwrap();
+        assert_eq!(last_block.block_number, last_block_load.block_number);
+    }
+
```

Comment:
There is no test which actually tests if some transactions are added to a latest block (and a new pending block created) by means of this periodic block generation.

Reply:
new test with transactions was added 6a422215dfe1affe0d080e31573f7b090596c59b